### PR TITLE
Profile with lighthouse

### PR DESCRIPTION
**Pull request by saulshanabrook**
This adds profiling page load time with [Lighthouse](https://github.com/GoogleChrome/lighthouse). We can use lighthouse to collect data data on how different Webpack changes affect end user load time. 

Fixes https://github.com/jupyterlab/jupyterlab/issues/4825

### CODE REVIEW

**Comment by saulshanabrook** [comment]
@gnestor I expect this would be useful for https://github.com/jupyterlab/jupyterlab/issues/5862, to see how lazy loading different modules affects initial page load times.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -196,6 +196,183 @@ directly. You can also use `jlpm test --namePattern=<regex>` to specify specific
...
+
+```bash
+jlpm run lighthouse --view
+```
+
+![](./docs/source/images/lighthouse-screenshot.png)
+
+### Using throttling
+
+Lighthouse reccomends using the system level [`comcast`](https://github.com/tylertreat/comcast) tool to throttle your network connection
```
Comment (mimischi) [inline CONTRIBUTING.md]:
```suggestion
Lighthouse recommends using the system level [`comcast`](https://github.com/tylertreat/comcast) tool to throttle your network connection
```

**Comment by bollwyvl** [comment]
Really awesome stuff!

Haven't been able to dig too much into it, but would it be straightforward
to pull out the raw logs as JSON and put them someplace where we can do
long-term or cross-branch comparisons?

**Comment by saulshanabrook** [comment]
@mimischi thank you 😁

**Comment by saulshanabrook** [comment]
> Haven't been able to dig too much into it, but would it be straightforward to pull out the raw logs as JSON and put them someplace where we can do long-term or cross-branch comparisons?

That would make sense! Ideally we could do this in CI so we could compute this for every commit. 

I would like to merge this first, so that we can start using it manually at least.

**Comment by telamonian** [comment]
I did 20 page refresh profiling runs by hand this morning using devtools. This sounds like a very welcome alternative.

**Comment by saulshanabrook** [comment]
This is ready for a review.

**Pull request merged**